packages/core/test/financial-templates/perpetual-multiparty/PerpetualPositionManager.js

```diff
@@ -1694,6 +1694,39 @@ contract("PerpetualPositionManager", function(accounts) {
     assert(await didContractThrow(positionManager.redeem({ rawValue: "16" }, { from: sponsor })));
   });
 
+  it("Gulp edge cases", async function() {
+    // Gulp does not revert if PfC and collateral balance are both 0
+    await positionManager.gulp();
+
+    // Send 1 wei of excess collateral to the contract.
+    await collateral.transfer(positionManager.address, "1", { from: sponsor });
+
+    // Gulp reverts if PfC is 0 but collateral balance is > 0.
+    assert(await didContractThrow(positionManager.gulp()));
+
+    // Create a position to gulp.
+    await collateral.approve(positionManager.address, toWei("100000"), { from: sponsor });
+    await tokenCurrency.approve(positionManager.address, toWei("100000"), { from: sponsor });
+    await positionManager.create({ rawValue: toWei("10") }, { rawValue: "20" }, { from: sponsor });
+
+    // Gulp does not do anything if the intermediate calculation (collateral balance / PfC) has precision loss.
```

Comment:
Nice test, this is really interesting. If I understand right the multiplier wont change unless the excess collateral is significant enough in size? Its possible that there could be some dust left in the contract in that case, not that it would be a problem i dont think.

Reply:
>  If I understand right the multiplier wont change unless the excess collateral is significant enough in size? Its possible that there could be some dust left in the contract in that case, not that it would be a problem i dont think.

Correct, if the calculation (collateral balance / pfc) has dust truncates to 1e18, then the multiplier will not change. I think this is no worst than the other situations where dust can accumulate. What's nice is that as the dust accumulates, `gulp()` can be used to absorb it once it reaches a certain threshold. Without `gulp`, the dust would be frozen collateral locked in the contract and not part of PfC.